### PR TITLE
Update release docs

### PR DESCRIPTION
**Pull request by saimn**
### Description

This pull request is to address ...

Fixes #<Issue Number>

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: docs/development/releasing.rst [hunk 1]
```diff
@@ -218,6 +222,21 @@ the release branch.
 Ensure continuous integration and intensive tests pass
 ------------------------------------------------------
 
+Update ``.github/workflows/ci_workflows.yml`` so that pushes on the release
```
[the remark this answers is not in the document]
Comment (saimn) [reply docs/development/releasing.rst:225]: On main the CI is triggered only for pushed on the main branch:
https://github.com/astropy/astropy/blob/269d7b81f31295351fd970ac0692e1cf2629174d/.github/workflows/ci_workflows.yml#L3-L9

[the remark this answers is not in the document]
Comment (pllim) [reply docs/development/releasing.rst:225]: Oh, okay. Maybe I was confused from  #12832

**Review by pllim — APPROVED** [review]
LGTM. I'll just trust you on the content since you actually did the release. Thanks!

File: docs/development/releasing.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (astrofrog) [reply docs/development/releasing.rst:225]: Huh why do we restrict the CI to the main branch? Can't we just build on all branches?

[the remark this answers is not in the document]
Comment (pllim) [reply docs/development/releasing.rst:225]: 🤷

**Pull request merged**